test/xpu_api/random/device_tests/bernoulli_distribution_device_test.pass.cpp

```diff
@@ -36,98 +35,178 @@ main()
 #if TEST_DPCPP_BACKEND_PRESENT && TEST_UNNAMED_LAMBDAS
 
     sycl::queue queue(exception_handler);
-    int err = 0;
-
-    // testing oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>> oneapi::dpl::linear_congruential_engine
-    std::cout << "---------------------------------------------------------------------" << std::endl;
-    std::cout << "bernoulli_distribution<sycl::vec<bool, 1>> linear_congruential_engine" << std::endl;
-    std::cout << "---------------------------------------------------------------------" << std::endl;
-    err = device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<std::uint32_t,  a, c, m>>(queue);
-#if TEST_LONG_RUN
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 16>,  a, c, m>>(queue);
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 8>,  a, c, m>>(queue);
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 4>,  a, c, m>>(queue);
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 3>,  a, c, m>>(queue);
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 2>,  a, c, m>>(queue);
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 1>,  a, c, m>>(queue);
-#endif // TEST_LONG_RUN
-    EXPECT_TRUE(!err, "Test FAILED");
+    if (TestUtils::has_type_support<double>(queue.get_device()))

```

Comment:
Probably it is better to change it like this:
if (!TestUtils::has_type_support<double>(queue.get_device()))
    return TestUtils::done();
    
Instead of such huge if{} 
It is up to you

Reply:
I've used functionality from Dmitriy's branch, so I suggest leaving it as it not to duplicate similar functions

Reply:
It seems Pavel's suggestion is not related to using functionality from #461. It is about following this approach:
```c++
if (!TestUtils::has_type_support(queue.get_device()))
    return TestUtils::done(0);
// all the test cases ...
```
Instead of that:
```c++
if (!TestUtils::has_type_support(queue.get_device()))
{
    //all the test cases ...
}
else
    return TestUtils::done(0);

```
Please, note that  I added `0`. It means that the exit message will be `skipped` instead of `done`.

Reply:
I believe Pavel's suggestion was to follow this approach:
```c++
if (!TestUtils::has_type_support(queue.get_device()))
    return TestUtils::done(0);
// all the test cases...
```
instead of this one:
```c++
if (!TestUtils::has_type_support(queue.get_device()))
{
    // all the test cases
}
else
    return TestUtils::done(0);
```
Please, note that I added `0`. The test will report `skipped` status instead of `done` since all the test cases have been skipped.

Reply:
Got it, but suggested approach doesn't work. Looks like compiler tries to JIT all kernel after if condition, so we need to have else branch to skip it